File: src/main/java/luminia/backend/models/Task.java
package luminia.backend.models;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;

import java.time.Duration;
import java.time.LocalDateTime;
import java.util.List;

@Entity(name = "Task")
@Table(name = "tasks")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "tasks_gen")
    @SequenceGenerator(name = "tasks_gen", sequenceName = "tasks_seq", allocationSize = 1)
    @Column(name = "id", nullable = false)
    private Long id;
    @Column(name = "display_name", unique = true)
    private String displayName;
    private String description;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "course_id")
    private Course course;
    private List<Long> attachments;
    @Column(name = "private_attachments")
    private List<Long> privateAttachments;
    private TaskType task;

    public enum TaskType {
        LECTURE, TASK
    }
}

File: frontend/src/services/remote/task.ts
import type Attachment from './attachment'
import type Course from './course'
import type List from './list'

export default class Task {
    public id: number
    public displayName: string
    public attachments: List<Attachment>
    public task: string
    public course: Course
    public status: string

    constructor(
        id: number,
        displayName: string,
        attachments: List<Attachment>,
        task: string,
        course: Course,
        status: string
    ) {
        this.id = id
        this.displayName = displayName
        this.attachments = attachments
        this.task = task
        this.course = course
        this.status = status
    }
}